packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts

```diff
@@ -20,18 +20,26 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
     protocol: req.protocol,
   } as AuthenticationInput
 
-  const authResult = await service.validateCallback(authProvider, authData)
+  const authResult = await service.validateCallback(auth_provider, authData)
 
-  const { success, error, authUser, location } = authResult
-  if (location) {
-    res.redirect(location)
-    return
-  }
+  const { success, error, authUser, successRedirectUrl } = authResult
 
   if (success) {
     const { jwt_secret } = req.scope.resolve("configModule").projectConfig
-    const token = jwt.sign(authUser, jwt_secret)
-    return res.status(200).json({ token })
+
+    const user = { ...authUser }
+    const token = jwt.sign(user, jwt_secret)
+
+    if (successRedirectUrl) {
+      const url = new URL(successRedirectUrl!)
+      url.searchParams.append("auth_token", token)
+
+      const redirectUrl = `${url}`
+
+      return res.redirect(redirectUrl.toString())
```

Comment:
```suggestion
      return res.redirect(redirectUrl)
```
nit

Reply:
Or just do `url.toString()` which I think is better.